src/SFML/Graphics/SpriteBatch.cpp

```diff
@@ -0,0 +1,266 @@
+{
+    bool unsupportedType = type != PrimitiveType::TriangleFan && type != PrimitiveType::TriangleStrip &&
+                           type != PrimitiveType::Triangles;
+
+    if (unsupportedType)
+    {
+        err() << "SpriteBatch supports only triangle-based primitive types.";
+        return;
+    }
+
+    if (count == 0)
+        return;
+
+    if (m_batchMode == BatchMode::Immediate)
+    {
+        // Immediate is the same as using a RenderTarget directly,
+        // so we don't use VertexBuffer here
+        m_renderTarget->draw(vertices, count, type, {m_blendMode, m_transform * transform, texture, m_shader});
+        return;
+    }
+
+    // Split into triangles if needed
+
+    if (type == PrimitiveType::TriangleStrip)
+    {
+        for (std::size_t i = 2; i < count; i++)
+        {
+            pushTriangle(vertices[i - 2], vertices[i - 1], vertices[i], transform, texture, depth);
+        }
+    }
+
+    else if (type == PrimitiveType::TriangleFan)
+    {
+        for (std::size_t i = 2; i < count; i++)
+        {
+            pushTriangle(vertices[0], vertices[i - 1], vertices[i], transform, texture, depth);
+        }
+    }
+
+    else // PrimitiveType::Triangles
+    {
+        for (std::size_t i = 2; i < count; i += 3)
+        {
+            pushTriangle(vertices[i - 2], vertices[i - 1], vertices[i], transform, texture, depth);
+        }
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::end()
+{
+    if (!m_batchActive)
+    {
+        err() << "A batch must be active to be able to end it.";
+        return;
+    }
+
+    if (m_batchMode == BatchMode::Deferred)
+    {
+        std::size_t triangleCount = m_triangleInfos.size();
+
+        // Batch based on texture change
+        std::size_t    start       = 0;
+        std::size_t    next        = 0;
+        const Texture* lastTexture = nullptr;
+
+        while (next < triangleCount)
+        {
+            const Texture* nextTexture = m_triangleInfos[next].texture;
+            if (nextTexture != lastTexture)
+            {
+                sendVertices(&m_vertices[start * 3], (next - start) * 3, lastTexture);
+
+                lastTexture = nextTexture;
+                start       = next;
+            }
+
+            next++;
+        }
+
+        // Deal with leftovers
+        if (start != m_triangleInfos.size())
+        {
+            sendVertices(&m_vertices[start * 3], (next - start) * 3, lastTexture);
+        }
+    }
+
+    else if (m_batchMode == BatchMode::DepthSort || m_batchMode == BatchMode::TextureSort)
+    {
+        std::vector<std::size_t> indices(m_triangleInfos.size());
+        std::size_t              triangleCount = m_triangleInfos.size();
+
+        for (std::size_t i = 0; i < triangleCount; i++)
+            indices[i] = i;
+
+        if (m_batchMode == BatchMode::TextureSort)
+        {
+            const auto& comp = [&](const std::size_t a, const std::size_t b)
```

Comment:
What are the implications of binding a lambda to a `const&`? I've only ever see it bound to a const value.

Reply:
I'm not sure, but I don't think there's much difference. I'll change it to regular `const` though.